### PR TITLE
luminous: class rbd.Image discard----OSError: [errno 2147483648] error discarding region

### PR DESCRIPTION
**Pull request by shinobu-x**
http://tracker.ceph.com/issues/22190

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

**Comment by theanalyst** [comment]
non clean cherry-pick, scheduling in a qe suite, needs @dillaman approval after clean run

File: PendingReleaseNotes [hunk 1]
```diff
@@ -38,3 +38,16 @@
   make the number of PGs per OSD exceed the configured
   ``mon_max_pg_per_osd`` limit.  The option can be adjusted if it
   is really necessary to create a pool with more PGs.
+
```
Comment (dillaman) [inline PendingReleaseNotes]: Nit: remove this since it looks unrelated

File: PendingReleaseNotes [hunk 2]
```diff
@@ -38,3 +38,16 @@
...
+* There was a bug in the PG mapping behavior of the new *upmap*
+  feature. If you made use of this feature (e.g., via the `ceph osd
+  pg-upmap-items` command), we recommend that all mappings be removed (via
+  the `ceph osd rm-pg-upmap-items` command) before upgrading to this
+  point release.
+
+13.0.0
```
Comment (dillaman) [inline PendingReleaseNotes]: Nit: change this to 12.2.X (whatever the next release is) or add it to the relevant section if one already exists.

**Comment by yuriw** [comment]
[rbd => wip-yuri2-testing-2018-01-24-1620-luminous](https://trello.com/c/tD5oVOpi/242-rbd-wip-yuri2-testing-2018-01-24-1620-luminous)

File: PendingReleaseNotes (hunk 2 above)
Comment (theanalyst) [reply PendingReleaseNotes]: ack.. scheduled tests anyway since the comments were only on the doc section which we can amend before merging
